moq-pub/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::Context;
+use clap::Parser;
+use tokio::task::JoinSet;
+
+mod session_runner;
+use session_runner::*;
+
+mod media_runner;
+use media_runner::*;
+
+mod log_viewer;
+use log_viewer::*;
+
+mod media;
+use media::*;
+
+mod cli;
+use cli::*;
+
+// TODO: clap complete
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+	env_logger::init();
+
+	let args = Cli::parse();
+
+	let config = Config {
+		addr: args.bind_address,
+		uri: args.uri,
+	};
+
+	let mut media = Media::new().await?;
+	let session_runner = SessionRunner::new(config).await?;
+	let mut log_viewer = LogViewer::new(session_runner.get_incoming_receivers().await).await?;
+	let mut media_runner = MediaRunner::new(
```

Comment:
nit: that's a lot of async constructors and awaits.

Reply:
True, I should see if some of these might be better to just be synchronous. I guess maybe all of these could be given how I'm using them here. 🤷‍♂️

Reply:
Actually, there are reasons this is simpler - mainly because I want to share some code in these constructors with fns that I _do_ want to be async later on. Could maybe do some more clean up later, but I'm not going to do that now.